### ISSUE TITLE
ban-single-arg-parens doesn't work with return type declarations

### ISSUE THREAD

**Issue by felixfbecker** [palantir/tslint#2237]
### Bug Report

- __TSLint version__: 4.4.2
- __TypeScript version__: 2.2.0
- __Running TSLint via__: (pick one) anything

#### TypeScript code being linted

```ts
const fn = (param): ReturnType {
  // whatever
}
```

with `tslint.json` configuration:

```json
		"arrow-parens": [true, "ban-single-arg-parens"],
```

#### Actual behavior
```
Parentheses are prohibited around the parameter in this single parameter arrow function
```
If you remove the parenthesis, TypeScript will not be able to compile the file. Parenthesis are required if you declare a return type.

#### Expected behavior
No error if a return type is declared
